demos/rag_agentic/notebooks/extras/mcp_toolgroup_management.ipynb

```diff
@@ -0,0 +1,365 @@
+   "id": "e41f432f-50b2-4a14-9ef0-c6e007c7ddb4",
+   "metadata": {},
+   "source": [
+    "## Register an MCP tool group \n",
+    "\n",
+    "Next, we register a new MCP tool group using `client.toolgroups.register(...)`.\n",
+    "\n",
+    "Ensure that the provided MCP endpoint is valid and accessible."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "125ad514-cf22-41fe-b89e-43203d21bb6b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Register MCP tools\n",
+    "mcp_custom_url = os.getenv(\"CUSTOM_MCP_SERVER_URL\")\n",
+    "if \"mcp::custom_tool\" not in registered_toolgroups:\n",
+    "    client.toolgroups.register(\n",
+    "        toolgroup_id=\"mcp::custom_tool\",\n",
+    "        provider_id=\"model-context-protocol\",\n",
+    "        mcp_endpoint={\"uri\":mcp_custom_url}\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "af7c6df2-1299-4d7e-8aa4-cd34f898a13f",
+   "metadata": {},
+   "source": [
+    "### Verify Registered tools\n",
+    "After registering a new MCP tool group, verify its presence by listing all currently registered tool groups using:\n",
+    "\n",
+    "```python\n",
+    "client.tools.list()\n",
+    "``` "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "id": "df189aaa-eba6-4522-8d2a-c02ab6ee40e8",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Registered Toolgroups:\n",
+      " - builtin::code_interpreter\n",
+      " - mcp::github\n",
+      " - mcp::openshift\n",
+      " - builtin::websearch\n",
+      " - builtin::rag\n",
+      " - mcp::slack\n",
+      " - mcp::custom_tool\n",
+      " - mcp::ansible\n"
+     ]
+    }
+   ],
+   "source": [
+    "toolgroups = client.tools.list()\n",
+    "registered_toolgroups = set(t.toolgroup_id for t in toolgroups)\n",
+    "\n",
+    "print(\"Registered Toolgroups:\")\n",
+    "for tg in registered_toolgroups:\n",
+    "    print(f\" - {tg}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f34d9262-34b2-43fa-808d-85ffccb23e62",
+   "metadata": {},
+   "source": [
+    "You should now see the custom tool group included in the list of registered tool groups."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "51094a16-f6e0-40cd-ae2a-5d85f73253b0",
+   "metadata": {},
+   "source": [
+    "##  View Tools in a Tool group"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d1b61ac7-7fea-4b18-ae6a-6060789d4f1f",
+   "metadata": {},
+   "source": [
+    "Let us inspect the tools available within a specific tool group, such as `mcp::custom_tool`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "id": "ca91c478-58f3-446e-a29d-010935b156f3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def list_tools_in_group(toolgroup_id):\n",
+    "    tools = client.tools.list(toolgroup_id=toolgroup_id)\n",
+    "    return [t.identifier for t in tools]"
```

Comment:
Does this need to be a function? Its only used once.?  Might it be simpler to just do this instead? 
```suggestion
    "    tools = client.tools.list(toolgroup_id="mcp::custom_tool")\n",
    "    tool_names = [t.identifier for t in tools]"
```